lib/log.c

```diff
@@ -65,8 +65,5 @@ _log (log_level level, const char *file, unsigned lineno, const char *fmt, ...)
     /* Print the user supplied message */
     vfprintf (stream, fmt, argptr);
 
-    /* always add a new line so the user doesn't have to */
-    fprintf (stream, "\n");
```

Comment:
Hi @martinezjavier,
Instead of removing the new line here and adding it to every invocation of the log macro
can we add new set macros that don't add the '\n'. The user can then choose if \n should be printed or not
bu using different macros. What about something like this below. Same effect, but much less churn:

diff --git a/lib/log.c b/lib/log.c
index 8078a00..1d4e690 100644
--- a/lib/log.c
+++ b/lib/log.c
@@ -42,7 +42,7 @@ get_level_msg (log_level level)
 }
 
 void
-_log (log_level level, const char *file, unsigned lineno, const char *fmt, ...)
+_log (log_level level, const char *file, unsigned lineno, bool nl, const char *fmt, ...)
 {
 
     /* Skip printing messages outside of the log level */
@@ -62,8 +62,8 @@ _log (log_level level, const char *file, unsigned lineno, const char *fmt, ...)
     /* Print the user supplied message */
     vfprintf (stderr, fmt, argptr);
 
-    /* always add a new line so the user doesn't have to */
-    fprintf (stderr, "\n");
+    if (nl)
+        fprintf (stderr, "\n");
 
     va_end(argptr);
 }
diff --git a/lib/log.h b/lib/log.h
index 32f43ec..5e477f3 100644
--- a/lib/log.h
+++ b/lib/log.h
@@ -11,7 +11,8 @@ typedef enum log_level log_level;
  *
  * Use this to log all error conditions.
  */
-#define LOG_ERR(fmt, ...) _log(log_level_error, __FILE__, __LINE__, fmt, ##__VA_ARGS__)
+#define LOG_ERR(fmt, ...) _log(log_level_error, __FILE__, __LINE__, 1, fmt, ##__VA_ARGS__)
+#define LOG_ERR_NONL(fmt, ...) _log(log_level_error, __FILE__, __LINE__, 0, fmt, ##__VA_ARGS__)
 
 /*
  * Prints an warning message. The fmt and variadic arguments mirror printf.
@@ -19,7 +20,8 @@ typedef enum log_level log_level;
  * Use this to log a warning. A warning is when something is wrong, but it is not a fatal
  * issue.
  */
-#define LOG_WARN(fmt, ...) _log(log_level_warning, __FILE__, __LINE__, fmt, ##__VA_ARGS__)
+#define LOG_WARN(fmt, ...) _log(log_level_warning, __FILE__, __LINE__, 1, fmt, ##__VA_ARGS__)
+#define LOG_WARN_NONL(fmt, ...) _log(log_level_warning, __FILE__, __LINE__, 0, fmt, ##__VA_ARGS__)
 
 /*
  * Prints an informational message. The fmt and variadic arguments mirror printf.
@@ -27,7 +29,8 @@ typedef enum log_level log_level;
  * Informational messages are only shown when verboseness is increased. Valid messages
  * would be debugging type messages where additional, extraneous information is printed.
  */
-#define LOG_INFO(fmt, ...) _log(log_level_verbose, __FILE__, __LINE__, fmt, ##__VA_ARGS__)
+#define LOG_INFO(fmt, ...) _log(log_level_verbose, __FILE__, __LINE__, 1, fmt, ##__VA_ARGS__)
+#define LOG_INFO_NONL(fmt, ...) _log(log_level_verbose, __FILE__, __LINE__, 0, fmt, ##__VA_ARGS__)
 
 enum log_level
 {
@@ -45,6 +48,6 @@ bool
 log_set_level (log_level level);
 
 void
-_log (log_level level, const char *file, unsigned lineno, const char *fmt, ...) __attribute__ ((format (printf, 4, 5)));
+_log (log_level level, const char *file, unsigned lineno, bool nl, const char *fmt, ...) __attribute__ ((format (printf, 5, 6)));
 
 #endif /* SRC_LOG_H_ */